src/k2/indexer/IndexerInterface.h

```diff
@@ -22,18 +22,63 @@ Copyright(c) 2020 Futurewei Cloud
 */
 
 #pragma once
-#include <k2/common/Common.h>
+#include <map>
+#include <deque>
+#include <utility>
 
+#include <k2/common/Common.h>
+#include <k2/dto/Collection.h>
+#include <k2/dto/K23SI.h>
 namespace k2
 {
 //
 //  K2 internal MVCC representation
 //
-template<typename ValueType>
-struct VersionedTreeNode {
-    uint64_t version;
-    ValueType value;
-    std::unique_ptr<VersionedTreeNode<ValueType>> next;
+
+template<typename IndexerType, typename ValueType>
```

Comment:
This sort of template args would require a bunch of redundant type specifications which may also lead to some compile issues at best, or runtime bugs at worst.
Could we use the map as template argument instead?
e.g. usage:
Indexer<std::map, dto::Key, dto::Record> idxr;
idxr.find(dto::Key{"a","b"});